core/templates/dev/head/domain/suggestion/SuggestionThreadObjectFactory.ts

```diff
@@ -17,82 +17,91 @@
  */
 
-require('domain/suggestion/SuggestionObjectFactory.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('SuggestionThreadObjectFactory', [
-  'SuggestionObjectFactory', function(SuggestionObjectFactory) {
-    var SuggestionThread = function(
-        status, subject, summary, originalAuthorName, lastUpdated, messageCount,
-        threadId, suggestion) {
-      this.status = status;
-      this.subject = subject;
-      this.summary = summary;
-      this.originalAuthorName = originalAuthorName;
-      this.lastUpdated = lastUpdated;
-      this.messageCount = messageCount;
-      this.threadId = threadId;
-      this.suggestion = suggestion;
-      this.messages = [];
-    };
+import { Suggestion, SuggestionObjectFactory } from
+  'domain/suggestion/SuggestionObjectFactory';
 
-    // TODO(ankita240796): Remove the bracket notation once Angular2 gets in.
-    /* eslint-disable dot-notation */
-    SuggestionThread['createFromBackendDicts'] = function(
-    /* eslint-enable dot-notation */
-        suggestionThreadBackendDict, suggestionBackendDict) {
-      var suggestion;
-      if (suggestionBackendDict.suggestion_type ===
-          'edit_exploration_state_content') {
-        suggestion = SuggestionObjectFactory.createFromBackendDict(
-          suggestionBackendDict);
-      }
-      return new SuggestionThread(
-        suggestionThreadBackendDict.status, suggestionThreadBackendDict.subject,
-        suggestionThreadBackendDict.summary,
-        suggestionThreadBackendDict.original_author_username,
-        suggestionThreadBackendDict.last_updated,
-        suggestionThreadBackendDict.message_count,
-        suggestionThreadBackendDict.thread_id, suggestion);
-    };
-
-    SuggestionThread.prototype.setMessages = function(messages) {
-      this.messages = messages;
-    };
-
-    SuggestionThread.prototype.getMessages = function() {
-      return this.messages;
-    };
+class SuggestionThread {
+  status: string;
+  subject: string;
+  summary: string;
+  originalAuthorName: string;
+  lastUpdated: number;
+  messageCount: number;
+  threadId: string;
+  suggestion: Suggestion;
+  messages: Array<{text: string}>;
+  constructor(
+      status: string, subject: string, summary: string,
+      originalAuthorName: string, lastUpdated: number, messageCount: number,
+      threadId: string, suggestion: Suggestion) {
+    this.status = status;
+    this.subject = subject;
+    this.summary = summary;
+    this.originalAuthorName = originalAuthorName;
+    this.lastUpdated = lastUpdated;
+    this.messageCount = messageCount;
+    this.threadId = threadId;
+    this.suggestion = suggestion;
+    this.messages = [];
+  }
 
-    SuggestionThread.prototype.isSuggestionHandled = function() {
-      return this.suggestion ? this.suggestion.status !== 'review' : null;
-    };
+  setMessages(messages: Array<{text: string}>): void {
```

Comment:
I am not sure of the type here. isn't messages an array of objects? I thought it had other data like message author, last updated time, etc. Could you please confirm?

Reply:
In the test file, 
```
let messages = [{
      text: 'message1'
    }, {
      text: 'message2'
    }];
```
This code is present. I deduced the type from this. I'll change it to any for now(with a Todo).

Reply:
Oh I see. They can have other elements too, but I think they are all strings (Not sure). To check, with the type you have written here, can you go to the feedback tab and create a thread, and add messages to it? After that refresh the page and it should still be visible. If there aren't any console errors, I think it should be ok to have this as the type.

Reply:
Hey Nithesh, I am unable to view the message object. I have created a thread.
![Screenshot 2019-12-17 at 12 25 01 AM](https://user-images.githubusercontent.com/30312043/70934442-f1df0f00-2063-11ea-9931-e36810c77e00.png)

Reply:
I am guessing that it is because last_updated is a float. I think you can try with the type as Array of {text: any} and that might work

Reply:
Ok Added any and a todo.

Reply:
Done